File: app/src/main/java/com/p1/uber/provides/DriverProvider.java
package com.p1.uber.provides;

import com.google.android.gms.tasks.Task;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.p1.uber.modelos.Drive;

public class DriverProvider {

    DatabaseReference mDatabase;

    public DriverProvider(){
        mDatabase = FirebaseDatabase.getInstance().getReference().child("Users").child("Drivers");
    }
    public Task<Void> create(Drive driver){
        return mDatabase.child(driver.getId()).setValue(driver);

    }

    public  DatabaseReference getDriver (String idDriver){
        return mDatabase.child(idDriver);
    }
}

File: app/src/main/java/com/p1/uberfares/provides/DriverProvider.kt
package com.p1.uberfares.provides

import com.google.android.gms.tasks.Task
import com.google.firebase.database.DatabaseReference
import com.google.firebase.database.FirebaseDatabase
import com.p1.uberfares.modelos.Drive

class DriverProvider {
    var mDatabase: DatabaseReference

    init {
        mDatabase = FirebaseDatabase.getInstance().reference.child("Users").child("Drivers")
    }

    fun create(driver: Drive): Task<Void> {
        return mDatabase.child(driver.id).setValue(driver)
    }

    fun getDriver(idDriver: String?): DatabaseReference {
        return mDatabase.child(idDriver!!)
    }
}